### ISSUE TITLE
no colors

### ISSUE THREAD

**Issue by JakobA-88** [joao7yt/CSHUE#5]
Hey. I have a lot of hue bulbs in my house. In my office I have RGB, but other places in the house I have only white. The program only allows me to change from warm to cold white, maybe because of the other only white bulbs.. How can I fix the color problem? I have only selected the bulbs in my office.
Hope you can help :D

**Comment by joao7yt** [comment]
Interesting... is the app thinking the rgb bulbs are also only temperature bulbs?

**Comment by JakobA-88** [comment]
yes. it shows u as warm/cold white bulbs only. I figured the bulbs in one of my lamps in the office is from IKEA (still rgb), the others is HUE. Is it possible that the IKEA (RGB) bulb, is making any trouble?

**Comment by joao7yt** [comment]
It’s possible... could you try to remove the bulb from your hub using the official hue app to see if that fixes?

**Comment by chaalss** [comment]
Hi ! Did you solve the issue? I've got a RGB ledstrip with a GLEDOPTO driver. On my hue app I can set all RGB colours, but on CSHUE I can only change to warm/cold white.

Thank you men!!

**Comment by joao7yt** [comment]
Is GLEDOPTO a brand? Do you own any Philips Hue lamp?

**Comment by chaalss** [comment]
Yes, sorry for bad explanation. 
I have a philips hue rgb bulb working well with CSHUE. I added a RGB strip with a GLEDOPTO driver, which is Hue compatible, but not from Philips brand. It is working well with Hue app, changing colours etc

In CSHUE, I can see the bulb and de strip, but only allows me to change RGB on the bulb. In the strip I can only change white temperature.

**Comment by joao7yt** [comment]
Okay, I would need to get to know how is these custom (non philips) lights handled by the philips hue api... Do you understand how to work with apis? If you don’t, I would have to create a special version of CSHUE and send to you through here. This version would create a txt file with the information I need... then you would send that to me.

**Comment by chaalss** [comment]
I really don't know how to use the API, but if you are "wasting" too much time only for my problem don't worry. If you still wanna try it, I will send you the txt file :)

**Comment by joao7yt** [comment]
I decided that a tutorial would be better.

Try these steps:

1. Access **https://discovery.meethue.com** on your browser and there will be the **ip** of your Bridge. Copy that.
2. Access **https://PutTheCopiedIpHere/debug/clip.html** replacing that part with the **Bridge ip**.
3. In that page, put **/api** in the **URL** field and **{"devicetype":"testing#desktop"}** in **Message Body**.
4. Now you have to **press your Bridge button** first and then click **POST**.
5. In **Command Response** there will be a **success** message followed by a key. Copy that key.
6. Now put **/api/PutTheCopiedKeyHere/lights** in the **URL** field, leave the **Message Body** field empty and then click **GET**.
7. Copy all the **Command Response** and paste it here for me.

If you have trouble, ask me about it...

**Comment by chaalss** [comment]
Thank you, below is the command response. What I see is that the non-philips lighstrip is not controlled the same way with "colorgamut". The way it's controlled is with "xy" parameter. The text is a blue light

{
	"3": {
		"state": {
			"on": false,
			"bri": 207,
			"hue": 8731,
			"sat": 106,
			"effect": "none",
			"xy": [
				0.436,
				0.404
			],
			"ct": 331,
			"alert": "select",
			"colormode": "hs",
			"mode": "homeautomation",
			"reachable": true
		},
		"swupdate": {
			"state": "noupdates",
			"lastinstall": "2020-04-02T09:26:56"
		},
		"type": "Extended color light",
		"name": "Techo",
		"modelid": "LCA001",
		"manufacturername": "Signify Netherlands B.V.",
		"productname": "Hue color lamp",
		"capabilities": {
			"certified": true,
			"control": {
				"mindimlevel": 200,
				"maxlumen": 800,
				"colorgamuttype": "C",
				"colorgamut": [
					[
						0.6915,
						0.3083
					],
					[
						0.17,
						0.7
					],
					[
						0.1532,
						0.0475
					]
				],
				"ct": {
					"min": 153,
					"max": 500
				}
			},
			"streaming": {
				"renderer": true,
				"proxy": true
			}
		},
		"config": {
			"archetype": "ceilinground",
			"function": "mixed",
			"direction": "omnidirectional",
			"startup": {
				"mode": "safety",
				"configured": true
			}
		},
		"uniqueid": "00:17:88:01:06:7a:f5:1d-0b",
		"swversion": "1.65.9_hB3217DF",
		"swconfigid": "BD38721C",
		"productid": "Philips-LCA001-5-A19ECLv6"
	},
	"4": {
		"state": {
			"on": true,
			"bri": 154,
			"hue": 40448,
			"sat": 242,
			"effect": "none",
			"xy": [
				0.1412,
				0.1296
			],
			"ct": 153,
			"alert": "select",
			"colormode": "xy",
			"mode": "homeautomation",
			"reachable": true
		},
		"swupdate": {
			"state": "notupdatable",
			"lastinstall": "2020-04-16T18:31:45"
		},
		"type": "Extended color light",
		"name": "luces despacho",
		"modelid": "GL-C-008",
		"manufacturername": "GLEDOPTO",
		"productname": "Extended color light",
		"capabilities": {
			"certified": false,
			"control": {
				"colorgamuttype": "other",
				"ct": {
					"min": 0,
					"max": 65535
				}
			},
			"streaming": {
				"renderer": false,
				"proxy": false
			}
		},
		"config": {
			"archetype": "huelightstrip",
			"function": "mixed",
			"direction": "omnidirectional"
		},
		"uniqueid": "00:12:4b:00:1e:fe:61:38-0b",
		"swversion": "2.0.0"
	}
}


--------------------

green light:
"state": {
			"on": true,
			"bri": 154,
			"hue": 16128,
			"sat": 5,
			"effect": "none",
			"xy": [
				0.2528,
				0.672
			],

----------------
red light
"state": {
			"on": true,
			"bri": 154,
			"hue": 2048,
			"sat": 250,
			"effect": "none",
			"xy": [
				0.6826,
				0.3041
			],

**Comment by joao7yt** [comment]
So, currently I'm using the information given by the API about the capabilities of the light to determine if I will show the color wheel or the temperature wheel to the user.

CSHUE is specifically looking for **"colorgamut"**, inside **"control"**, inside **"capabilities"**... In your GLEDOPTO light, there's no **"colorgamut"** there. But if the HUE app can know it's a color light, then I should too.

For this I will need someone with a Temperature only light to do the same what you did so I can see the difference between the API response, changing the method I use to know if it's a color or temperature light.

**Comment by NanobugGG** [comment]
I have this issue, or at least a similiar one.
I've launched the CSHUE, set it all up (at least I think), but it's not reacting to anything.
When I set the colors, it does change it on the LED strip.
I can't figure out why.

Can you or someone else help?

**Comment by joao7yt** [comment]
> I have this issue, or at least a similiar one.
> I've launched the CSHUE, set it all up (at least I think), but it's not reacting to anything.
> When I set the colors, it does change it on the LED strip.
> I can't figure out why.
> 
> Can you or someone else help?

Did you create the .cfg in the right folder?

**Comment by NanobugGG** [comment]
> > I have this issue, or at least a similiar one.
> > I've launched the CSHUE, set it all up (at least I think), but it's not reacting to anything.
> > When I set the colors, it does change it on the LED strip.
> > I can't figure out why.
> > Can you or someone else help?
> 
> Did you create the .cfg in the right folder?

Maybe.... What's the right folder?

**Comment by joao7yt** [comment]
> > > I have this issue, or at least a similiar one.
> > > I've launched the CSHUE, set it all up (at least I think), but it's not reacting to anything.
> > > When I set the colors, it does change it on the LED strip.
> > > I can't figure out why.
> > > Can you or someone else help?
> > 
> > 
> > Did you create the .cfg in the right folder?
> 
> Maybe.... What's the right folder?

steamapps\common\Counter-Strike Global Offensive\csgo\cfg\

**Comment by NanobugGG** [comment]
> > > > I have this issue, or at least a similiar one.
> > > > I've launched the CSHUE, set it all up (at least I think), but it's not reacting to anything.
> > > > When I set the colors, it does change it on the LED strip.
> > > > I can't figure out why.
> > > > Can you or someone else help?
> > > 
> > > 
> > > Did you create the .cfg in the right folder?
> > 
> > 
> > Maybe.... What's the right folder?
> 
> steamapps\common\Counter-Strike Global Offensive\csgo\cfg\

I didn't realize that. I'll do it right away!

**Comment by NanobugGG** [comment]
Mother of....
Thank you!

I even tried deleting it all and start over. I didn't see or think of that it had to be placed in the cfg folder of CS:GO.

**Comment by joao7yt** [comment]
you are welcome. i will delete these posts in a few minutes because it doesn't belong here anyway. have fun with the app :)
